### PR TITLE
Log every inventory action

### PR DESCRIPTION
**Pull request by Emojigit**
This PR adds logging functions to every inventory action. The existing logging functions for chests are also improved. This PR is ready for review.

### CODE REVIEW

**Comment by S-S-X** [comment]
```
Error → spec/api_spec.lua @ 6
CNC API
api.lua:386: attempt to index global 'technic' (a nil value)
```

Just use `technic_cnc.use_technic` to check if `technic` namespace is usable with CNC.
Technic is _optional_ dependency for CNC, it also works without technic mod. Both cases are tested: with and without technic mod.

**Comment by OgelGames** [comment]
The `technic.machine_on_inventory_*` functions don't need to return functions, you can use `minetest.get_node(pos).name` to get the node name.

**Comment by Emojigit** [comment]
For some reason, [`pipeworks` is requiring `fakelib`](https://github.com/mt-mods/technic/actions/runs/9364488850/job/25777590925?pr=360#step:4:30)... That's why `mtt` failed.

**Comment by BuckarooBanzay** [comment]
> For some reason, [`pipeworks` is requiring `fakelib`](https://github.com/mt-mods/technic/actions/runs/9364488850/job/25777590925?pr=360#step:4:30)... That's why `mtt` failed.

Yeah, you can ignore that, i'll create another PR to add that dependency  :+1:

File: technic/machines/register/common.lua [hunk 1]
```diff
@@ -190,3 +190,39 @@ function technic.machine_inventory_move(pos, from_list, from_index,
...
+	local nodename = minetest.get_node(pos).name
+	minetest.log("action", string.format(
+		"%s moved item from %s[%u] to %s[%u] of %s at %s",
+		player:is_player() and player:get_player_name() or "A mod",
+		from_list, from_index,
+		to_list, to_index,
+		nodename,
+		minetest.pos_to_string(pos)
+	))
+end
```
Comment (OgelGames) [inline technic/machines/register/common.lua:228]: The log messages don't need to be that verbose.
```suggestion
function technic.machine_on_inventory_put(pos, listname, index, stack, player)
	minetest.log("action", string.format("%s puts %s into %s at %s",
		player:get_player_name(),
		stack:to_string(),
		minetest.get_node(pos).name,
		minetest.pos_to_string(pos)
	))
end

function technic.machine_on_inventory_take(pos, listname, index, stack, player)
	minetest.log("action", string.format("%s takes %s from %s at %s",
		player:get_player_name(),
		stack:to_string(),
		minetest.get_node(pos).name,
		minetest.pos_to_string(pos)
	))
end

function technic.machine_on_inventory_move(pos, from_list, from_index, to_list, to_index, count, player)
	local stack = minetest.get_meta(pos):get_inventory():get_stack(to_list, to_index)
	minetest.log("action", string.format("%s moves %s in %s at %s",
		player:get_player_name(),
		stack:to_string(),
		minetest.get_node(pos).name,
		minetest.pos_to_string(pos)
	))
end
```

**Review by OgelGames — CHANGES_REQUESTED** [review]

**Comment by OgelGames** [comment]
Do you think it would be a good idea to handle non-player actions like `default` does? With all the extra logging, there's going to be a lot of spam from pipeworks moving items.

https://github.com/minetest/minetest_game/blob/511619253facd524df75ae44537d7b2c4e30a8db/mods/default/functions.lua#L729-L745

**Review by OgelGames — APPROVED** [review]

**Comment by OgelGames** [comment]
I've created a new issue about my suggestion, so I'm merging this.

**Pull request merged**